src/app/clusters/thread-network-diagnostics-server/thread-network-diagnostics-provider.cpp

```diff
@@ -0,0 +1,861 @@
+        }
+        else if (otRole == OT_DEVICE_ROLE_ROUTER)
+        {
+            routingRole = RoutingRoleEnum::kRouter;
+        }
+        else if (otRole == OT_DEVICE_ROLE_LEADER)
+        {
+            routingRole = RoutingRoleEnum::kLeader;
+        }
+        else if (otRole == OT_DEVICE_ROLE_CHILD)
+        {
+            otLinkModeConfig linkMode = otThreadGetLinkMode(otInst);
+
+            if (linkMode.mRxOnWhenIdle)
+            {
+                routingRole = RoutingRoleEnum::kEndDevice;
+#if CHIP_DEVICE_CONFIG_THREAD_FTD
+                if (otThreadIsRouterEligible(otInst))
+                {
+                    routingRole = RoutingRoleEnum::kReed;
+                }
+#endif
+            }
+            else
+            {
+                routingRole = RoutingRoleEnum::kSleepyEndDevice;
+            }
+        }
+
+        err = encoder.Encode(routingRole);
+    }
+    break;
+
+    case Attributes::NetworkName::Id: {
+        const char * networkName = otThreadGetNetworkName(otInst);
+        err                      = encoder.Encode(CharSpan::fromCharString(networkName));
+    }
+    break;
+
+    case Attributes::PanId::Id: {
+        uint16_t panId = otLinkGetPanId(otInst);
+        err            = encoder.Encode(panId);
+    }
+    break;
+
+    case Attributes::ExtendedPanId::Id: {
+        const otExtendedPanId * pExtendedPanid = otThreadGetExtendedPanId(otInst);
+        err                                    = encoder.Encode(Encoding::BigEndian::Get64(pExtendedPanid->m8));
+    }
+    break;
+
+    case Attributes::MeshLocalPrefix::Id: {
+        uint8_t meshLocaPrefix[OT_MESH_LOCAL_PREFIX_SIZE + 1] = { 0 }; // + 1  to encode prefix Len in the octstr
+
+        const otMeshLocalPrefix * pMeshLocalPrefix = otThreadGetMeshLocalPrefix(otInst);
+        meshLocaPrefix[0]                          = OT_IP6_PREFIX_BITSIZE;
+
+        memcpy(&meshLocaPrefix[1], pMeshLocalPrefix->m8, OT_MESH_LOCAL_PREFIX_SIZE);
+        err = encoder.Encode(ByteSpan(meshLocaPrefix));
+    }
+    break;
+
+    case Attributes::OverrunCount::Id: {
+        uint64_t overrunCount = 0; // mOverrunCount;
+        err                   = encoder.Encode(overrunCount);
+    }
+    break;
+
+    case Attributes::NeighborTable::Id: {
+        err = encoder.EncodeList([otInst](const auto & aEncoder) -> CHIP_ERROR {
+            constexpr uint16_t kFrameErrorRate100Percent   = 0xffff;
+            constexpr uint16_t kMessageErrorRate100Percent = 0xffff;
+
+            otNeighborInfo neighInfo;
+            otNeighborInfoIterator iterator = OT_NEIGHBOR_INFO_ITERATOR_INIT;
+
+            while (otThreadGetNextNeighborInfo(otInst, &iterator, &neighInfo) == OT_ERROR_NONE)
+            {
+                Structs::NeighborTableStruct::Type neighborTable;
+                app::DataModel::Nullable<int8_t> averageRssi;
+                app::DataModel::Nullable<int8_t> lastRssi;
+
+                if (neighInfo.mAverageRssi == OT_RADIO_RSSI_INVALID)
```

Comment:
Suggestion to reduce the number of lines
`(neighInfo.mAverageRssi == OT_RADIO_RSSI_INVALID) ? ... : ...;`

Reply:
cannot use .setNull in this format
compiler throws
` error: second operand to the conditional operator is of type 'void', but the third operand is neither a throw-expression nor of type 'void'`

will keep it as is